src/controllers/currentWarningsController.js

```diff
@@ -0,0 +1,155 @@
+        return res.status(200).send({ error: 'warning already exists' });
+      }
+
+      const newWarningDescription = new currentWarnings();
+      newWarningDescription.warningTitle = newWarning;
+      newWarningDescription.activeWarning = activeWarning;
+      newWarningDescription.isPermanent = isPermanent;
+
+      warnings.push(newWarningDescription);
+      await newWarningDescription.save();
+
+      return res.status(201).send({ newWarnings: warnings });
+    } catch (error) {
+      return res.status(401).send({ message: error.message });
+    }
+  };
+
+  const editWarningDescription = async (req, res) => {
+    try {
+      const { editedWarning } = req.body;
+
+      const id = editedWarning._id;
+
+      const warnings = await currentWarnings.find({});
+
+      if (warnings.length === 0) {
+        return res.status(400).send({ message: 'no valid records' });
+      }
+
+      const lowerCaseWarning = editedWarning.warningTitle.toLowerCase();
+      const testWarning = checkIfSpecialCharacter(lowerCaseWarning);
+
+      if (testWarning) {
+        return res.status(200).send({
+          error: 'Warning cannot have special characters as the first letter',
+        });
+      }
+
+      if (checkForDuplicates(lowerCaseWarning, warnings)) {
+        return res.status(200).send({ error: 'warning already exists try a different name' });
+      }
+
+      await currentWarnings.findOneAndUpdate(
+        { _id: id },
+        [{ $set: { warningTitle: lowerCaseWarning.trim() } }],
+        { new: true },
+      );
+
+      res.status(201).send({ message: 'warning description was updated' });
+    } catch (error) {
+      res.status(401).send({ message: error.message || error });
+    }
+  };
+  const updateWarningDescription = async (req, res) => {
+    try {
+      const { warningDescriptionId } = req.params;
+
+      await currentWarnings.findOneAndUpdate(
+        { _id: warningDescriptionId },
+        [{ $set: { activeWarning: { $not: '$activeWarning' } } }],
+        { new: true },
+      );
+
+      res.status(201).send({ message: 'warning description was updated' });
+    } catch (error) {
+      res.status(401).send({ message: error.message || error });
+    }
+  };
+
+  const deleteWarningDescription = async (req, res) => {
+    try {
+      const { warningDescriptionId } = req.params;
+      const documentToDelete = await currentWarnings.findById(warningDescriptionId);
+
+      await currentWarnings.deleteOne({
+        _id: mongoose.Types.ObjectId(warningDescriptionId),
+      });
+
+      const deletedDescription = documentToDelete.warningTitle;
+
+      await userProfile.updateMany(
+        {
+          'warnings.description': deletedDescription,
+        },
+        {
+          $pull: {
+            warnings: { description: deletedDescription },
+          },
+        },
+      );
+
+      res.status(200).send({
+        message: 'warning description was successfully deleted and user profiles updated',
```

Comment:
`200` already indicates this is a successful operation, so no need for messages.